File: arkts_static_test/02_LexicalElements/ets_cases/2.5_Tokens/java_runtime_test/TokensBasicValidationTest.java
/*
* Copyright (c) 2021-2026 Huawei Device Co., Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
*/
// Java Runtime Test - 对应 2.5 Tokens 独立 Token 验证
// 测试重点：各类 Token 的基本使用语法正确性
// 注意：Java 不完全等价于 ArkTS Token，设定为概念映射测试

public class TokensBasicValidationTest {
    // Identifiers (映射到 Java 命名)
    private String simpleIdentifier = "myVar";
    private String identifierWithDigits = "var2";
    private String identifierWithDollar = "var$Name";

    // Keywords mapping (概念映射)
    // Note: Java 没有 let/const，用 static final/局部变量模拟
    private static final int LET_KEYWORD = 10;   // 映射 let 常量
    private static final int IF_KEYWORD = 20;    // 映射 if 条件
    private static int WHILE_LOOP = 0;          // 映射 while 循环
    private int CLASS_INSTANCE = 0;             // 映射 class 实例化

    // Operators mapping (概念映射 - Java有对应运算符)
    private int arithmeticOp = 10 + 20 + 30;          // + 运算
    private int comparisonOp = 5 < 10 && 10 > 5;      // < && > 比较
    private int logicalOp = true && false || true;     // && || 逻辑
    private int assignmentOp = 10; assignmentOp += 5;  // = += 赋值

    public static void main(String[] args) {
        int count = 0;

        // 测试 2.5 章节：各类 Token 基本语法有效性
        System.out.println("[Java] Token Testing: Font-end Validation");

        // Identifiers 测试
        String id1 = "simpleIdentifier"; count++;
        assert id1.equals("simpleIdentifier") : "Identifier 123 failed";
        System.out.println("[Java] Identifier test: PASSED");

        String id2 = "identifierWithDigits"; count++;
        assert id2.equals("identifierWithDigits") : "Identifier with digits failed";
        System.out.println("[Java] Identifier (with digits) test: PASSED");

        String id3 = "identifierWithDollar"; count++;
        assert id3.equals("identifierWithDollar") : "Identifier with dollar failed";
        System.out.println("[Java] Identifier (with $) test: PASSED");

        // Keywords test (概念映射)
        int k1 = LET_KEYWORD; count++;
        assert k1 == 10 : "Keyword 'let' mapping test failed";
        System.out.println("[Java] Keyword mapping (let): PASSED");

        int k2 = IF_KEYWORD; count++;
        assert k2 == 20 : "Keyword 'if' mapping test failed";
        System.out.println("[Java] Keyword mapping (if): PASSED");

        // 运算符测试
        int a1 = arithmeticOp; count++;
        assert a1 == 60 : "Arithmetic operator test failed, got " + a1;
        System.out.println("[Java] Arithmetic operator ( + - * / ) test: PASSED (60)");

        int c1 = comparisonOp; count++;
        assert c1 == 1 : "Comparison operator test failed, got " + c1;
        System.out.println("[Java] Comparison operator (== != < > <= >=) test: PASSED");

        boolean l1 = logicalOp; count++;
        assert l1 == true : "Logical operator test failed, got " + l1;
        System.out.println("[Java] Logical operator (&& || !) test: PASSED");

        int as1 = assignmentOp; count++;
        assert as1 == 15 : "Assignment operator test failed, got " + as1;
        System.out.println("[Java] Assignment operator (= += -= *= /=) test: PASSED (15)");

        // 字面量测试
        int i1 = 42; count++;
        assert i1 == 42 : "Integer literal test failed, got " + i1;
        System.out.println("[Java] Integer literal test: PASSED (42)");

        double f1 = 3.14; count++;
        assert f1 == 3.14 : "Float literal test failed, got " + f1;
        System.out.println("[Java] Float literal test: PASSED (3.14)");

        boolean b1 = true; count++;
        assert b1 == true : "Boolean literal test failed, got " + b1;
        System.out.println("[Java] Boolean literal test: PASSED (true)");

        // Token 组合测试
        int tokenExpr = 10 + 20 + 30; count++;
        assert tokenExpr == 60 : "Token combination test failed, got " + tokenExpr;
        System.out.println("[Java] Token combination test: PASSED");

        System.out.println("=== Java Tokens Basic Validation Test PASSED ===");
        System.out.println("Total assertions passed: " + count);
    }
}

File: arkts_static_test/02_LexicalElements/ets_cases/2.5_Tokens/swift_runtime_test/TokensBasicValidationTest.swift
/*
* Copyright (c) 2021-2026 Huawei Device Co., Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
*/
// Swift Runtime Test - 对应 2.5 Tokens 独立 Token 验证
// 测试重点：各类 Token 的基本使用语法正确性

class TokensBasicValidationTest {
    // Identifiers (映射到 Swift 命名)
    var simpleIdentifier = "myVar"
    var identifierWithDigits = "var2"
    var identifierWithDollar = "var$Name"
    private let CONST_KEYWORD = 10   // 映射 const
    private let LET_KEYWORD = 20      // 映射 let

    // Methods representing token concepts
    func testArithmeticOperators() -> Bool {
        let result = 10 + 20 + 30
        return result == 60
    }

    func testComparisonOperators() -> Bool {
        let a = 5
        let b = 10
        return a < b && b > a
    }

    func testLogicalOperators() -> Bool {
        return true && false || true
    }

    func testAssignmentOperators() -> Int {
        var result = 10
        result += 5
        result -= 3
        result *= 2
        result /= 1
        return result
    }

    @main
    struct Main {
        static func main() {
            print("=== Swift Tokens Basic Validation Test ===")
            print("")

            var assertionsPassed = 0

            // 测试 2.5 章节：各类 Token 基本语法有效性

            // Identifiers 测试
            assert("simpleIdentifier" == "simpleIdentifier") { "Identifier 1 failed" }
            assertionsPassed += 1
            print("[Swift] Identifier test: PASSED")

            assert("identifierWithDigits" == "identifierWithDigits") { "Identifier 2 failed" }
            assertionsPassed += 1
            print("[Swift] Identifier (with digits) test: PASSED")

            assert("identifierWithDollar" == "identifierWithDollar") { "Identifier 3 failed" }
            assertionsPassed += 1
            print("[Swift] Identifier (with $) test: PASSED")

            // Keywords test (概念映射)
            assert(CONST_KEYWORD == 10) { "Keyword 'const' mapping test failed" }
            assertionsPassed += 1
            print("[Swift] Keyword mapping (const): PASSED")

            assert(LET_KEYWORD == 20) { "Keyword 'let' mapping test failed" }
            assertionsPassed += 1
            print("[Swift] Keyword mapping (let): PASSED")

            // 运算符测试
            let a1 = TokensBasicValidationTest().testArithmeticOperators()
            assert(a1 == true) { "Arithmetic operator test failed" }
            assertionsPassed += 1
            print("[Swift] Arithmetic operator ( + - * / ) test: PASSED")

            let c1 = TokensBasicValidationTest().testComparisonOperators()
            assert(c1 == true) { "Comparison operator test failed" }
            assertionsPassed += 1
            print("[Swift] Comparison operator (== != < > <= >=) test: PASSED")

            let l1 = TokensBasicValidationTest().testLogicalOperators()
            assert(l1 == true) { "Logical operator test failed" }
            assertionsPassed += 1
            print("[Swift] Logical operator (&& || !) test: PASSED")

            let assignmentResult = TokensBasicValidationTest().testAssignmentOperators()
            assert(assignmentResult == 12) { "Assignment operator test failed, got \(assignmentResult)" }
            assertionsPassed += 1
            print("[Swift] Assignment operator (= += -= *= /=) test: PASSED (12)")

            // 字面量测试
            let intLiteral = 42
            assert(intLiteral == 42) { "Integer literal test failed" }
            assertionsPassed += 1
            print("[Swift] Integer literal test: PASSED (42)")

            let floatLiteral = 3.14
            assert(floatLiteral == 3.14) { "Float literal test failed" }
            assertionsPassed += 1
            print("[Swift] Float literal test: PASSED (3.14)")

            let boolLiteral = true
            assert(boolLiteral == true) { "Boolean literal test failed" }
            assertionsPassed += 1
            print("[Swift] Boolean literal test: PASSED (true)")

            // Token 组合测试
            let comboExpr = 10 + 20 - 5
            assert(comboExpr == 25) { "Token combination test failed, got \(comboExpr)" }
            assertionsPassed += 1
            print("[Swift] Token combination test: PASSED (25)")

            print("")
            print("=== Swift Tokens Basic Validation Test PASSED ===")
            print("Total assertions passed: \(assertionsPassed)")
        }
    }
}
